sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1967,6 +1969,9 @@ class Analyzer(override val catalogManager: CatalogManager)
     override def apply(plan: LogicalPlan): LogicalPlan = {
       val externalFunctionNameSet = new mutable.HashSet[FunctionIdentifier]()
       plan.resolveExpressions {
+        case f @ UnresolvedFunction(NonSessionCatalogAndIdentifier(_, _), _, _, _, _) =>
+          // no-op if this is from a v2 catalog
```

Comment:
shall we add the v2 API to check function existence?

Reply:
Yes looking at [SPARK-19737](https://issues.apache.org/jira/browse/SPARK-19737) I think this will be very useful (although the checking won't be complete since `bind` could still fail). Do you mind if I add a TODO here and do it separately later?

Reply:
SGTM